### DISCUSSION TITLE
[@types/ember-data] Return value of Store.query is incorrect

### DISCUSSION THREAD

**Discussion by lolmaus** [DefinitelyTyped/DefinitelyTyped#55562] [Issues with a @types package]
The actual return value of `store.query()` contains a property `meta`.

Such property exists in type `ManyArray` and, by inherence, `PromiseManyArray`.

But `query` is typed to return `PromiseArray` that does not contain `meta`. 

CC @chriskrycho @dfreeman @jamescdavis @mike-north

[a comment by an automated account is not shown]

**Comment by chriskrycho** [comment]
@lolmaus do you mind asking this over on the ember-cli-typescript repo instead? We track everything like this there (although we are admittedly behind!) because there’s just too much traffic/noise on the DT repo!

**Reply by lolmaus** [reply]
https://github.com/typed-ember/ember-cli-typescript/issues/1446

**Reply by chriskrycho** [reply]
Thank you! Replied there. Summary for anyone who happens across this question: it's unclear what the path forward is, because the docs say we shouldn't be returning `ManyArray` or `PromiseManyArray` at all, but Data is in something of a transitional state. We’ll figure it out there.
